src/webpage/src/app.rs

```diff
@@ -567,6 +573,55 @@ impl App {
         }
     }
 
+    fn deal_with_parameters_message(&mut self, message: String) {
+        let Ok(message_json) = serde_json::from_str::<serde_json::Value>(&message) else {
+            log::error!("Failed to parse Vehicles message: {message}");
+            return;
+        };
+
+        // We only deal with a single vehicle for now
+        if !message_json.as_object().unwrap().contains_key("1") {
+            return;
+        }
+        let parameter = message_json["1"].as_object().unwrap();
+        for (key, value) in parameter {
+            self.parameters.insert(key.to_string(), value.clone());
+        }
+    }
+
+    fn process_parameters_websocket(&mut self) {
+        use ewebsock::{WsEvent, WsMessage};
+
+        loop {
+            match self.parameters_receiver.try_recv() {
+                Some(WsEvent::Message(WsMessage::Text(message))) => {
+                    self.deal_with_parameters_message(message)
+                }
+                Some(WsEvent::Closed) => {
+                    log::error!("Parameters WebSocket closed");
+                    (self.parameters_sender, self.parameters_receiver) =
+                        connect_websocket(CONTROL_VEHICLES_WEBSOCKET_PATH).unwrap();
+
+                    break;
+                }
+                Some(WsEvent::Error(message)) => {
+                    log::error!("Parameters WebSocket error: {message}");
+                    (self.parameters_sender, self.parameters_receiver) =
+                        connect_websocket(CONTROL_PARAMETERS_WEBSOCKET_PATH).unwrap();
+
+                    break;
+                }
+                Some(WsEvent::Opened) => {
+                    log::info!("Parameters WebSocket opened");
+                }
+                something @ Some(_) => {
+                    log::trace!("Parameters WebSocket got an unexpected event: {something:#?}");
+                }
+                None => break,
+            }
+        }
+    }
```

Comment:
wait, is this in the right commit?

Reply:
yes, parameters are now received via websocket